### PR TITLE
Detailed notes on Windows installer versioning issues

### PR DESCRIPTION
**Pull request by hoffmang9**
Microsoft Windows requires that any executable have three and only three fields for version number.

Quoting from: https://docs.microsoft.com/en-us/windows/win32/msi/productversion?WT.mc_id=email
The first field is the major version and has a maximum value of 255.
 The second field is the minor version and has a maximum value of 255. 
 The third field is called the build version or the update version and has a maximum value of 65,535.

### CODE REVIEW

**Review by AmineKhaldi — DISMISSED** [review]
LGTM after PR #10027 lands.

File: build_scripts/installer-version.py
Comment (altendky) [inline build_scripts/installer-version.py:14]: I don't see that quote there.  Rather, I see this.

> Note that Windows Installer uses only the first three fields of the product version. If you include a fourth field in your product version, the installer ignores the fourth field.

But it does say that one of the first three has to change for an update so I guess that's the bit where this bubbles up and is important.  Doesn't seem like it needs to drive the odd file name on Windows, but that's separate from this PR.

Comment (hoffmang9) [reply build_scripts/installer-version.py:14]: Somehow I'm causing confusion. 

Our installers in Windows create a .exe. If you try to run a .exe that has either 1.1.2.1 or 1.1.2dev226 the installer will segfault. 

The windows installer get their versions here as they are translated from setuptools_DCM into mostly semver versions.

Comment (altendky) [reply build_scripts/installer-version.py:14]: I did not test via the installer as squirrel did not like the non-semver four segment version and I did not take the time to find the rest of this winding path...  but the issue at least does not apply to all executables in all cases as the comment being added to the code states.  Also, semver does have support for pre-release indicators.

image

I do not mean to argue that it is a priority to improve our development version reporting on Windows.  It isn't clear to me what the boundaries of the failure mode are so I'm not sure if or when this three-segment restriction would no longer be applicable.  I would expect a clarification here to express, with whatever degree of detail is remembered, either the specific scenario that was actually observed to be an issue, or, to link to documentation that directly confirms the statement.

**Pull request closed without merging**
